api/nodes/views.py

```diff
@@ -228,12 +294,62 @@ def get_default_odm_query(self):
         query = base_query & permission_query
         return query
 
-    # overrides ListCreateAPIView
+    # overrides ListBulkCreateUpdateDestroyAPIView
     def get_queryset(self):
-        query = self.get_query_from_request()
-        return Node.find(query)
+        # For bulk requests, queryset is formed from request body.
+        if isinstance(self.request.data, list):
+            query = Q('_id', 'in', [node['id'] for node in self.request.data])
+
+            user = self.request.user
+            if user.is_anonymous():
+                auth = Auth(None)
+            else:
+                auth = Auth(user)
 
-    # overrides ListCreateAPIView
+            nodes = Node.find(query)
+            for node in nodes:
+                if not node.can_edit(auth):
```

Comment:
This appears to be mixing permissions logic into the view, which we try to avoid.

Can we separate this out into a permissions class instead?